pkg/provider/aws/client.go

```diff
@@ -166,344 +157,319 @@ func addProxyConfigToSessionOptConfig(config *aws.Config) {
 		if err != nil {
-			return nil, fmt.Errorf("could not load config file: %v", err)
+			return nil, fmt.Errorf("could not load config file: %w", err)
+		}
+		cfg, err = config.LoadDefaultConfig(context.TODO(), config.WithRegion(region), config.WithSharedConfigProfile(profile), config.WithSharedConfigFiles([]string{absCfgPath}))
+	} else {
+		cfg, err = config.LoadDefaultConfig(context.TODO(), config.WithRegion(region), config.WithSharedConfigProfile(profile))
+		if err != nil {
+			return nil, fmt.Errorf("error loading aws config: %w", err)
 		}
-		opt.SharedConfigFiles = []string{absCfgPath}
 	}
 
-	sess := session.Must(session.NewSessionWithOptions(opt))
-	if _, err := sess.Config.Credentials.Get(); err != nil {
-		if aerr, ok := err.(awserr.Error); ok {
-			switch aerr.Code() {
-			case "NoCredentialProviders":
-				return nil, fmt.Errorf("could not create AWS session: %v", err)
-			default:
-				return nil, fmt.Errorf("could not create AWS session: %v", err)
-			}
-		}
+	addProxyConfigToSessionOptConfig(&cfg)
+
+	if _, err := cfg.Credentials.Retrieve(context.TODO()); err != nil {
+		return nil, fmt.Errorf("failed to retrieve AWS credentials: %w", err)
 	}
 
-	return sess, nil
+	return &cfg, nil
 }
 
 // NewAwsClient creates an AWS client with credentials in the environment
 func NewAwsClient(profile, region, configFile string) (Client, error) {
-	sess, err := NewAwsSession(profile, region, configFile)
+	cfg, err := NewAwsConfig(profile, region, configFile)
 	if err != nil {
 		return nil, err
 	}
 
 	awsClient := &AwsClient{
-		iamClient:           iam.New(sess),
-		ec2Client:           ec2.New(sess),
-		stsClient:           sts.New(sess),
-		s3Client:            s3.New(sess),
-		servicequotasClient: servicequotas.New(sess),
-		orgClient:           organizations.New(sess),
-		ceClient:            costexplorer.New(sess),
-		resClient:           resourcegroupstaggingapi.New(sess),
-		cloudTrailClient:    cloudtrail.New(sess),
+		iamClient:           *iam.NewFromConfig(*cfg),
+		ec2Client:           *ec2.NewFromConfig(*cfg),
+		stsClient:           *sts.NewFromConfig(*cfg),
+		s3Client:            *s3.NewFromConfig(*cfg),
+		servicequotasClient: *servicequotas.NewFromConfig(*cfg),
+		orgClient:           *organizations.NewFromConfig(*cfg),
+		ceClient:            *costexplorer.NewFromConfig(*cfg),
+		resClient:           *resourcegroupstaggingapi.NewFromConfig(*cfg),
+		cloudTrailClient:    *cloudtrail.NewFromConfig(*cfg),
 	}
 
 	// Validate the creds
 	if _, err := awsClient.GetCallerIdentity(nil); err != nil {
-		if aerr, ok := err.(awserr.Error); ok {
-			switch aerr.Code() {
-			case "InvalidClientTokenId":
-				return nil, fmt.Errorf("`aws sts get-caller-identity --profile %s --region %s` failed - "+
-					"ensure the credentials in the profile are valid and the region is in the AWS partition you expect", profile, region)
-			default:
-				return nil, err
-			}
-		}
+		return nil, fmt.Errorf("error getting caller identity: %w", err)
```

Comment:
Emphasis that this changed, personally I'm ok with it, but so other reviewers are aware

Reply:
For context - https://github.com/openshift/osdctl/pull/411#discussion_r1268674566